rfc-119-merge-mainstream-browse-topics-and-taxons.md

```diff
@@ -0,0 +1,149 @@
+
+## Problem
+
+We currently maintain three seperate ways to categorise content items - mainstream browse pages, specialist topics and taxons. There is currently ongoing work to align the titles of mainstream browse and specialist topics to have equivalents in the taxonomy i.e. mainstream browse and specialist topics become a subset of taxons.
+
+Once this work is complete we want to maintain alignment and prevent divergence from taxons. We would have to put constraints in collections publisher and content tagger (or any future apps) to ensure that mainstream browse and specialist topics continue to be subsets of taxons. This would involve adding complex logic to the publishing system to deal with addition, changing or removal of these content item formats.
+
+## Proposal
+
+We propose merging the attributes of mainstream browse and specialist topics into taxon content items with the same `title`. 
+
+### Top level attributes
+
+The following attributes will continue to have the existing values in the taxon:
+- title
+- description
+- analytics_identifier
+- content_id
+- document_type
+- first_published_at
+- locale
+- phase
+- public_updated_at
+- publishing_scheduled_at
+- rendering_app
+- scheduled_publishing_delay_seconds
+- schema_name
+- updated_at
+- withdrawn_notice
+- publishing_request_id
+
+This is because they don't cause conflict or the mainstream or specialist topic value becomes redundant.
+
+The attribute `publishing_app` is `collections-publisher` for mainstream and specialist topics, however is `content-tagger` for taxons. We propose that it remains `content-tagger` as it still has the responsibility of creating and removing taxons, even though some sub properties (i.e. links and details) may be set by `collections-publisher`.
+
+Mainstream browse and specialist topic pages have different base paths to taxons. We propose that taxons keep their existing base paths. We could use new mainstream browse/specialist topic links within taxons to generate old base or find the appropriate taxon from an existing mainstream browse or specialist topic base path. Alternatively, we keep the old base paths as renamed attributes (e.g. mainstream_browse_base_path).
+
+### Link attributes
+
+Taxons continue to have the following links:
+- legacy_taxons (potentially remove once specialist topics are removed)
+- parent_taxons
+- available_translations
+
+Add from mainstream browse:
+- children (rename to mainstream_browse_children)
+- top_level_browse_pages (potentially could be dropped, as can be derived from path)
+- second_level_browse_pages (superseded by mainstream_browse_parent)
+
+Add from specialist topic:
+- children (rename to specialist_topic_children)
+- parent (rename to specialist_topic_parent)
```

Comment:
This might be my lack of knowledge of the taxonomy showing here... Taxons already model a parent/child relationship [through the `parent_taxons` link and `child_taxons` reverse link](https://github.com/alphagov/publishing-api/blob/0baad432aa0f573322bc2c28a10afdd968031ab4/lib/expansion_rules.rb#L40), rather than adding `children` and `parent` links, if we're merging them all into taxons, can we not use the existing `parent_taxons` and `child_taxons` links.

Reply:
So we'd be looking at having multiple hierarchies within the same taxons as the different sets of relationships are useful for different things ie. the currently hierarchy for taxonomy is good for emails subscriptions and organising taxons, but not so great at modelling navigation. So we'd bring over the mainstream browse hierarchy to serve that purpose.